app/Libraries/BBCodeFromDB.php

```diff
@@ -129,6 +129,54 @@ public function parseHeading($text)
         return $text;
     }
 
+    public function parseImagemap($text)
+    {
+        return preg_replace_callback(
+            '#(\[imagemap\].+?\[/imagemap\]\n?)#',
+            function ($m) {
+                return preg_replace_callback(
+                    '#\[imagemap\]\n(?<imageUrl>https?://.+)\n(?<links>(?:(?:[0-9.]+ ){4}(?:\#|https?://[^ ]+|mailto:[^ ]+)(?: .+)?\n)+)\[/imagemap\]\n?#',
+                    function ($map) {
+                        $links = array_map(
+                            fn ($rawLink) => explode(' ', $rawLink, 6),
+                            explode("\n", $map['links']),
+                        );
+                        array_pop($links); // remove the empty string from last newline
+
+                        $linksHtml = implode('', array_map(
+                            fn ($link) => tag($link[4] === '#' ? 'span' : 'a', [
+                                'class' => 'imagemap__link',
+                                'href' => $link[4],
+                                'style' => implode(';', [
+                                    "left: {$link[0]}%",
+                                    "top: {$link[1]}%",
+                                    "width: {$link[2]}%",
+                                    "height: {$link[3]}%",
+                                ]),
+                                'title' => $link[5] ?? '',
+                            ]),
+                            $links,
+                        ));
+
+                        $imageUrl = proxy_media($map['imageUrl']);
+                        $imageSize = fast_imagesize($imageUrl) ?? [0, 0];
```

Comment:
I think there needs to be something better than rendering a 0x0 image if it fails to get the size, or just skip rendering the image. It's also not apparent to the user if something goes wrong here, they just get an image that gets displayed at 0px